Sources/MapboxDirections/MatrixOptions.swift

```diff
@@ -0,0 +1,200 @@
+import Foundation
+import Turf
+
+/**
+ Options for calculating matrices from the Mapbox Matrix service.
+ */
+public class MatrixOptions: Codable {
+    // MARK: Creating a Matrix Options Object
+    /**
+     Initializes a matrix options object for matrices and a given profile identifier.
+     
+     - parameter waypoints: An array of `Waypoints` objects representing locations that will be in the matrix. The array should contain at least X coordinates and at most X coordinates. (Some profiles, such as `ProfileIdentifier`, [may have lower limits](https://docs.mapbox.com/api/navigation/matrix/#matrix-api-restrictions-and-limits).)
+     - parameter profileIdentifier: A string specifying the primary mode of transportation for the routes.
+     */
+    public init(waypoints: [Waypoint], profileIdentifier: ProfileIdentifier) {
+        self.waypoints = waypoints
+        self.profileIdentifier = profileIdentifier
+    }
+    
+    /**
+     A string specifying the primary mode of transportation for the contours.
+    */
+    public var profileIdentifier: ProfileIdentifier
+    
+    /**
+     An array of `Waypoints` objects representing locations that will be in the matrix.
+     */
+    public var waypoints: [Waypoint]
```

Comment:
Should we reuse `Waypoint` for the Matrix API? Judging from [the public API documentation](https://docs.mapbox.com/api/navigation/matrix/), many of the options in `Waypoint` don’t apply to the Matrix API, either as input or output.

Reply:
My thought was that using same `Waypoint` would be useful when after requesting the Matrix, user will pick these same waypoints to build an actual route.